src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -90,29 +92,29 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
 
         # Do stuff with DBS
         try:
+            # check if block exists in DBS
+            records = dbsApi.listBlocks(block_name=name)
+            # the records are shown in the following form
+            # [{'block_name': '/block/name#123'}]
+            if len(records) == 1 and records[0]['block_name'] == name:
+                # found that we have this block, i.e. no need to insert block anymore
+                logging.info("block %s already exist in DBS", name)
+                continue
```

Comment:
Before continue, don't we have to declare this block as injected, hence:
```
            results.put({'name': name, 'success': "uploaded"})
```
? Such that the agent updates the block status in the local database. Otherwise this block will keep showing up in the cycle, every time.

Reply:
yes, you are right, I added entry to the queue.